tools/embedded_data_analyzer.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/python3
+
+HELP_STRING = '''
+Tock embedded data analysis tool
+
+Generates a report of a Tock binary's embedded data usage in the form of a static HTML webpage.
+
+Prerequisite:
+    - The binary must be compiled for the RISC-V architecture.
+    - Turn off LTO. To do so, set lto to false in the top-level Cargo.toml file.
+
+To run the tool, in the Tock home directory invoke:
+
+    $ tools/embedded_data_analyzer.py -e <elf_path> -n <name> -o <output_directory>
+
+arguments:
+    elf_path         -- the relative path of the compiled elf file
+    name             -- a name for the binary to be displayed in the HTML page
+    output_directory -- where the user wishes the output to be stored
+
+You can also run the script with -h flag for help on what the arguments are.
+'''
+
+
+import re
+import sys
+import os
+from collections import defaultdict, namedtuple
+import argparse
+
+from print_tock_memory_usage import parse_mangled_name
+from embedded_data_visualizer import create_html_file
+
+OBJDUMP = 'riscv64-unknown-elf-objdump'
```

Comment:
Unfortunately every distro has a different name for this. Can we add a fallback checker like we have in [libtock-c](https://github.com/tock/libtock-c/blob/master/Configuration.mk#L124)

Reply:
I second this. It's unfortunate this does not seem to work with `llvm-objdump`, but many distros do indeed choose a different toolchain tuple. The list in `libtock-c` seems like a good start, or just adding an argument with a default value for this.

Reply:
Yes, I added the functionality to search the system path for known distributions of riscv objdump. As well as a fallback option for manually specifying which objdump to use.